Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,190 @@
+
+
+@_passThrough
+class Output:
+    """
+    Output specifies how to scatter data from a Python object into a table column.
+    """
+    def __init__(self, colName, scatterFunc, type=None):
+        """
+        Args:
+            colName         : name of the new column that will store results.
+            scatterFunc     : python function that determines how data is taken from a python object into a Deephaven table.
+            type (optional) : desired data type of the new output column. Must be one of int, long, double, float, boolean, char, String.
+        """
+        self.output = _Output_(colName, scatterFunc, type)
+
+    def __str__(self):
+        print("Column: " + self.output.getColName())
+        print("Type: " + self.output.getType())
+
+
+@_passThrough
+def _validate(inputs, outputs, table):
+    """
+    Ensures that all input columns exist in the table, and that no output column names already exist in the table.
+
+    Args:
+        inputs  : list of Inputs to validate.
+        outputs : list of Outputs to validate.
+        table   : table to check Input and Output columns against.
+
+    Raises:
+        ValueError : if at least one of the Input columns does not exist in the table.
+        ValueError : if at least one of the Output columns already exists in the table.
+    """
+
+    inputColumns = [input.input.getColNames()[i] for input in inputs for i in range(len(input.input.getColNames()))]
+
+    if not all(inputColumn in list(table.getMeta().getColumn("Name").getDirect()) for inputColumn in inputColumns):
+        raise ValueError("All Input columns must exist in the given table.")
+
+    if not outputs == None:
+        columns = [*set(inputColumns), *[output.output.getColName() for output in outputs]]
+
+        if len(columns) != len(set(columns)):
+            raise ValueError("Cannot use existing column names for Output columns.")
+
+    return
+
+
+@_passThrough
+def _verifyColumn(column, table):
+    """
+    Ensures that the given column is not present in the table, and modifies it if necessary until it is not in the table.
+
+    Args:
+        column : column name that should not exist in the table.
+        table  : table to check column against.
+
+    Returns:
+        the column name, modified if necessary so that it is not present in the table.
+    """
+
+    tableColumns = list(table.getMeta().getColumn("Name").getDirect())
+
+    if column not in tableColumns:
+        return column
+
+    else:
+        i = 0
+        while column in tableColumns:
+            column = column + str(i)
+
+            if column not in tableColumns:
+                return column
+            i += 1
+
+
+@_passThrough
+def eval(table=None, model_func=None, inputs=[], outputs=[], batch_size = None):
+    """
+    Takes relevant data from Deephaven table using inputs, converts that data to the desired Python type, feeds
+    it to model_func, and stores that output in a Deephaven table using outputs.
```

Comment:
Will your c-student friend with no familiarity with the code understand this?  Will someone coming to the code know that this is the key AI function?